### PR TITLE
Final

### PR DESCRIPTION
**Pull request by efratkohen**

### CODE REVIEW

File: hw5.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
+import numpy as np
+import pandas as pd
+import pathlib
+import matplotlib.pyplot as plt
+
```
Comment (ZviBaratz) [inline hw5.py:5]: Function signatures should have been kept as they were, including the typing. In order to include the typing as it was specified, you needed to add:
```python
from typing import Union, Tuple
```

File: hw5.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+class QuestionnaireAnalysis:
+    """
+    Reads and analyzes data generated by the questionnaire experiment.
+    Should be able to accept strings and pathlib.Path objects.
+    """
+
+    def __init__(self, data_fname):
+        self.data_fname = pathlib.Path(str(data_fname))
+        if self.data_fname.exists()==False:
```
Comment (ZviBaratz) [inline hw5.py:14]: Please consider:
```python
if not self.data_fname.is_file():
    raise ValueError("Invalid data file path!")
```

File: hw5.py [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+            raise ValueError("file is not exist")
+
+    def read_data(self):
+        """Reads the json data located in self.data_fname into memory, to
+        the attribute self.data.
+        """
+        with open(self.data_fname, "r") as f:
+            data = eval(f.read())
+        self.data=pd.DataFrame(data).replace('nan', np.NaN)
+        return self.data
```
Comment (ZviBaratz) [inline hw5.py:24]: Please consider:
```python
pd.read_json(self.data_fname)
```
Note: using `eval` is strongly discouraged.

File: hw5.py [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+    def show_age_distrib(self) -> [np.ndarray, np.ndarray]:
+        """Calculates and plots the age distribution of the participants.
+
+        Returns
+        -------
+        hist : np.ndarray
+        Number of people in a given bin
+        bins : np.ndarray
+        Bin edges
```
Comment (ZviBaratz) [inline hw5.py:34]: The description for the parameters and return values is an important part of the code! Please always try and include it with your code.

File: hw5.py (hunk 3 above)
Comment (HagaiHargil) [reply hw5.py:24]: @efratkohen Do you understand why using `eval` is almost never the right thing to do? Especially under these circumstances?

**Pull request closed without merging**